### PR TITLE
Exporter can soft pause and resume

### LINKED ISSUE #16344: Exporter Director can Soft Pause (And Resume)
Exporter can soft pause and resume. This means if the exporter is soft paused, we keep exporting the records without updating the exporter state. upon resuming The exporter state is updated with the position and metadata of the last exported record.

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

Exporter can soft pause and resume. This means if the exporter is soft paused, we keep exporting the records without updating the exporter state. upon resuming The exporter state is updated with the position and metadata of the last exported record.

The feature to persist the soft paused state will be implemented in this issue https://github.com/camunda/zeebe/issues/16642

## Related issues

closes https://github.com/camunda/zeebe/issues/16344

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
:rocket: Awesome. I like the behavior of consecutive calls of pause vs soft-pause, and resume always resumes full exporting, no matter if it was soft paused or hard paused. :+1: 

:x: I have only one main comment. The exporter director should be able to start with phase soft-paused if it was soft-paused before restart. You can do it in a follow up PR. If so please create an issue for it and link it here for transparency.

Other comments are minor - like improvements to tests and documentation. Please address them in this PR itself.

File: broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirector.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public ExporterDirector(final ExporterDirectorContext context, final boolean sho
     exportingRetryStrategy = new BackOffRetryStrategy(actor, Duration.ofSeconds(10));
     recordWrapStrategy = new EndlessRetryStrategy(actor);
     zeebeDb = context.getZeebeDb();
-    isPaused = shouldPauseOnStart;
+    isHardPaused = shouldPauseOnStart;
```
[the remark this answers is not in the document]
Comment (rodrigo-lourenco-lopes) [reply broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirector.java:101]: I will also implement this part in another PR for this issue https://github.com/camunda/zeebe/issues/16642

**Review by deepthidevaki — APPROVED** [review]
:rocket:

**Pull request merged**